--- a/src/sys/fs/hammer2/hammer2_freemap.c
+++ b/src/sys/fs/hammer2/hammer2_freemap.c
@@ -330,7 +330,8 @@
 			 * NOTE: bmap pointer is invalid if n >= FREEMAP_COUNT.
 			 */
 			n = start + count;
-			bmap = &chain->data->bmdata[n];
+			bmap = n < HAMMER2_FREEMAP_COUNT ?
+			    &chain->data->bmdata[n] : NULL;
 			if (n >= HAMMER2_FREEMAP_COUNT)
 				availchk = 0;
 			else if (bmap->avail)
@@ -368,7 +369,7 @@
 			 * NOTE: bmap pointer is invalid if n < 0.
 			 */
 			n = start - count;
-			bmap = &chain->data->bmdata[n];
+			bmap = n >= 0 ? &chain->data->bmdata[n] : NULL;
 			if (n < 0)
 				availchk = 0;
 			else if (bmap->avail)
